### PR TITLE
improve lilypond path preference initializer

### PR DESCRIPTION
**Pull request by nittka**
proposed solution for #102

### CODE REVIEW

File: org.elysium.ui/src/org/elysium/ui/compiler/preferences/CompilerPreferenceInitializer.java [hunk 1]
```diff
@@ -30,4 +35,42 @@ public void initializeDefaultPreferences() {
...
 	}
 
+
+	private String getDefaultLilypondPath() {
+		Object startParameterPath = System.getProperties().get("lilypond.path");//$NON-NLS-1$
+		if (startParameterPath != null) {
+			return startParameterPath.toString();
+		}
+
+		String userHome = System.getenv().get("HOME");//$NON-NLS-1$
```
Comment (thSoft) [inline org.elysium.ui/src/org/elysium/ui/compiler/preferences/CompilerPreferenceInitializer.java]: I'd rather take only the lilypond.path system property into account and generate the launch config in the Oomph setup with this system property set (like the test launch config).

**Comment by nittka** [comment]
Is there actually a (global operating) system property for the lilypond path? What I did was adding -Dlilypond.path=... to the launch configuration of the runtime Eclipse. The idea was that if someone added that to the eclipse.ini, that value should have the highest priority and be used for the preference initializer.
I agree that guessing the path from the user setup is somewhat magical. The intention here was as follows: Oomph allows recording preferences, but when *I* apply the defaults, the windows default path will always be the wrong one. So why not use the lilypond path I defined for the Elysium development setup.

If a user is not happy with the default, adding the start parameter to the eclipse.ini is a clean and sufficient way.

**Comment by nittka** [comment]
I have simplified the preference initializer according to the above discussion. When releasing the Elysium version including this pull request, it would make sense to adapt the documentation stating that adding -Dlilypond.path=... to the -vmargs section of the eclipse.ini allows specifying the default executable location.

**Comment by nittka** [comment]
second commit is a proposed fix for #104

**Comment by thSoft** [comment]
Cool, thanks!

**Pull request merged**